components/sections/NavigationBar/index.tsx

```diff
@@ -0,0 +1,105 @@
+import Hamburger from '@heroicons/react/24/solid/Bars3Icon';
+import XMark from '@heroicons/react/24/solid/XMarkIcon';
+import classNames from 'classnames';
+import Image from 'next/image';
+import { type FC, type ComponentProps, useState } from 'react';
+
+import LanguageDropdown from '@/components/Common/LanguageDropDown';
+import ThemeToggle from '@/components/Common/ThemeToggle';
+import NavItem from '@/components/sections/NavItem';
+import { BASE_PATH } from '@/next.constants.mjs';
+
+import style from './index.module.css';
+type NavItem = { text: string; href: string };
+
+type NavbarProps = {
+  navItems: NavItem[];
+  currentNavItem: string;
+  languages: ComponentProps<typeof LanguageDropdown>;
+  onThemeTogglerClick: () => void;
+};
+
+const NavigationBar: FC<NavbarProps> = ({
+  navItems,
+  currentNavItem,
+  languages,
+  onThemeTogglerClick,
+}) => {
+  const [showHamIcon, setShowHamIcon] = useState(true);
+  return (
+    <nav className={style.container}>
+      <div className={style.leftItems}>
+        <div className={style.nodeIconWrapper}>
+          <Image
+            className="dark:hidden"
+            height={24}
+            width={80}
+            src={`${BASE_PATH}/static/images/logos/horizontal-dark.svg`}
+            alt="Node.js"
+          />
+          <Image
+            className="hidden dark:block"
+            height={24}
+            width={80}
+            src={`${BASE_PATH}/static/images/logos/horizontal-light.svg`}
+            alt="Node.js"
+          />
+        </div>
+        <label
+          onClick={() => setShowHamIcon(prev => !prev)}
+          className={style.sidebarItemTogglerLabel}
+          htmlFor="sidebarItemToggler"
+        >
+          {showHamIcon ? (
```

Comment:
What if we have a simply silly map:

```tsx
// outside the component
const navInteractionIcons = {
  'show': Hamburger,
  'close': XMark,
}

// in the component
{navInteractionIcons[isOpen ? 'close' : 'show']}
```

And let's rename the state from `showHamIcon` to `isOpen`